### PR TITLE
Add OAuth 2.0 client for Twitch

### PR DESCRIPTION
**Pull request by angelospanag**
Added OAuth 2.0 client for Twitch along with tests and updated the examples in the `Provided clients` section of the documentation.

### CODE REVIEW

**Comment by Phifo** [comment]
I just implemented my own twitch oauth client because I didn't realize the existence of this PR. Now I can confirm my implementation was in the right direction, it's very similar to this. Thank you!

File: httpx_oauth/clients/twitch.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+from httpx_oauth.errors import GetIdEmailError
+from httpx_oauth.oauth2 import BaseOAuth2
+
+AUTHORIZE_ENDPOINT = "https://id.twitch.tv/oauth2/authorize"
+ACCESS_TOKEN_ENDPOINT = "https://id.twitch.tv/oauth2/token"
+REVOKE_TOKEN_ENDPOINT = "https://id.twitch.tv/oauth2/revoke"
+BASE_SCOPES = [
+    "user:read:email",
+    "user:read:follows",
```
Comment (frankie567) [inline httpx_oauth/clients/twitch.py:11]: Does all those scopes are required in a basic use case (i.e. get the id and email)? If not, I prefer to keep this list as short as possible, and let the end-user add its own set of scopes to fit their needs.

File: httpx_oauth/clients/twitch.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "user:manage:whispers",
+]
+
+"""
+If an ID or login name is not specified, the request to the user endpoint returns 
+information about the user specified in the included access token.
+
+https://dev.twitch.tv/docs/api/reference#get-users
+"""
+PROFILE_ENDPOINT = "https://api.twitch.tv/helix/users"
```
Comment (frankie567) [inline httpx_oauth/clients/twitch.py:22]: Have you tried this endpoint: https://id.twitch.tv/oauth2/userinfo ?

It seems more suited for we want to do here. Ref: https://dev.twitch.tv/docs/authentication/getting-tokens-oidc#getting-claims-information-from-an-access-token

File: httpx_oauth/clients/twitch.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+LOGO_SVG = """
```
Comment (frankie567) [inline httpx_oauth/clients/twitch.py:24]: For consistency with other logos, I prefer we use this version: https://github.com/gilbarbara/logos/blob/master/logos/twitch.svg?short_path=7cb962f

**Review by frankie567 — CHANGES_REQUESTED** [review]
Thank you for this PR 🙏 A Twitch client will be a valuable addition!

Looks good overall, I've added some minor comments 🙂

File: httpx_oauth/clients/twitch.py (hunk 3 above)
Comment (angelospanag) [reply httpx_oauth/clients/twitch.py:24]: No problem, fixed -> https://github.com/frankie567/httpx-oauth/pull/298/commits/dd071a1039206eac05800f52f682ba07921c250d

**Pull request closed without merging**